core-libs/setup/ssr/providers/ssr-providers.ts

```diff
@@ -4,12 +4,93 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { StaticProvider } from '@angular/core';
+import { Optional, SkipSelf, StaticProvider } from '@angular/core';
+import { INITIAL_CONFIG, PlatformConfig } from '@angular/platform-server';
 import { REQUEST } from '@nguniversal/express-engine/tokens';
 import { SERVER_REQUEST_ORIGIN, SERVER_REQUEST_URL } from '@spartacus/core';
 import { getRequestOrigin } from '../util/request-origin';
 import { getRequestUrl } from '../util/request-url';
 
+/**
+ * Returns the providers used for SSR and pre-rendering processes.
+ */
+export function provideSsrAndPrerendering(options?: {
+  /**
+   * Specify a domain (origin) from which the HTTP requests are being made.
+   * Should be without the trailing slash, e.g. "https://my.domain.com"
+   */
+  serverRequestOrigin?: string;
+}): StaticProvider[] {
+  return [
+    {
+      provide: SERVER_REQUEST_ORIGIN,
+      useFactory: (serverRequestOrigin?: string): string => {
```

Comment:
For readability of the providers array, I suggest to move functions to separate functions. Btw. we can use `inject()` instead of `deps` array, which has a better type safety.

Reply:
Thanks, good suggestion. I extracted the factories to separate files, please take a look.
Will proceed to testing, after we settle on https://github.com/SAP/spartacus/pull/16365#discussion_r1030243703